src/rgw/rgw_sal_dbstore.h

```diff
@@ -232,20 +232,74 @@ class DBNotification : public Notification {
       friend class DBStore;
   };
 
+  class DBPlacementTier: public PlacementTier {
+    DBStore* store;
+    RGWZoneGroupPlacementTier tier;
+  public:
+    DBPlacementTier(DBStore* _store, const RGWZoneGroupPlacementTier& _tier) : store(_store), tier(_tier) {}
+    virtual ~DBPlacementTier() = default;
+
+    virtual const std::string& get_tier_type() { return tier.tier_type; }
+    virtual const std::string& get_storage_class() { return tier.storage_class; }
+    virtual bool retain_head_object() { return tier.retain_head_object; }
+    RGWZoneGroupPlacementTier& get_rt() { return tier; }
```

Comment:
if all the backends are just forwarding values from `RGWZoneGroupPlacementTier`, do we really need a sal abstraction for `PlacementTier`? i'm not sure that `RGWZoneGroupPlacementTier` has any backend-specific stuff that we need to hide

Reply:
I did this as part of the module work, so RGWZoneGroupPlacementTier was living in code that was only compiled into stores.  I didn't want to move it, because it has an embedded RGWRZoneGroupPlacementTierS3, which pulls in other stuff, and it got complicated.  I don't have a strong opinion, but there's a lot of stuff in those two structs, when all we need in the API is a few fields.

Reply:
> I didn't want to move it, because it has an embedded RGWRZoneGroupPlacementTierS3, which pulls in other stuff, and it got complicated.

understood, but i do think this needs to happen eventually. the fields are there because they're necessary for the feature, and our API will eventually need to surface everything. i'd prefer not to bloat the SAL API surface for things like this, especially when all of the backends end up looking exactly the same. but if you see this as a stepping stone to further refactoring, that's fine too - maybe we can track that with `// TODO` comments

p.s. i just sent an email to the dev list titled 'rgw: zipper and API boundaries' that touches on this